tests/research/test_research_harness.py

```
@@ -0,0 +1,828 @@
+            "commit",
+            "-m",
+            "freeze",
+        ],
+        cwd=code_repo,
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    sha = _git(code_repo, "rev-parse", "HEAD")
+
+    def _raise_oserror(path):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_raise_oserror`

Add return type annotation: `NoReturn`

(ANN202)